### PR TITLE
Fix World Cleanup and Refactor as System Param

### LINKED ISSUE #131: Exiting Pause menu behaves unexpectedly
### Description

Exiting pause menu returns the player to the main menu, abandoning in progress gameplay.
This is not the desired behavior either for gameplay or debugging purposes. Ideally for gameplay exiting pause state would return the players to in progress session.
For debugging it is useful to not have the current world despawned while the pause is taking place, so that entities and components can be inspected using bevy-inspector-egui.

### To Reproduce

Pause the game, exit pause menu by pressing ESC key or Continue menu button.

### PR DESCRIPTION
**Pull request by zicklag**
Fixes #131

### CODE REVIEW

**Review by 64kramsystem — APPROVED** [review]
:+1:

File: src/main.rs [hunk 1]
```diff
@@ -211,7 +213,6 @@ fn main() {
         .add_startup_system(set_audio_channels_volume)
         .add_enter_system(GameState::InGame, play_level_music)
         .add_exit_system(GameState::InGame, stop_level_music)
-        .add_exit_system(GameState::InGame, clean_in_game_data)
```
Comment (odecay) [inline src/main.rs:214]: Is this the change which basically prevents cleanup from happening everytime `GameState::InGame` exits? This is the fix and the rest is mostly just cleanup correct?

**Review by odecay — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (zicklag) [reply src/main.rs:214]: Yep, this is the fix.

The rest of the code just makes the logic re-usable.

Comment (odecay) [reply src/main.rs:214]: Cool let's get it in

**Comment by zicklag** [comment]
bors merge

**Pull request merged**